srtcore/api.cpp

```diff
@@ -247,6 +291,95 @@ int CUDTUnited::cleanup()
    return 0;
 }
 
+SRTSOCKET CUDTUnited::generateSocketID()
```

Comment:
The generation of the socket ID is now widely changed because the socket ID will be also used for groups. Socket ID will now be using less bits as before, as one bit is now reserved to mark that the ID is of a group. New identifiers will come from the same pool, except that for the groups it will get this additional bit.